custom/icds_reports/utils/aggregation.py

```diff
@@ -1535,6 +1535,381 @@ def indexes(self, aggregation_level):
+            ('counsel_fp_methods', 'NULL'),
+            ('pregnant', 'ucr.pregnant'),
+            ('pregnant_all', 'ucr.pregnant_all'),
+            ('lactating', 'ucr.lactating'),
+            ('lactating_all', 'ucr.lactating_all'),
+            ('institutional_delivery_in_month', 'ucr.institutional_delivery_in_month'),
+            ('add', 'ucr.add'),
+            ('caste', 'ucr.caste'),
+            ('disabled', 'ucr.disabled'),
+            ('minority', 'ucr.minority'),
+            ('resident', 'ucr.resident'),
+            ('valid_in_month', 'ucr.valid_in_month'),
+            ('anc_in_month',
+             '( '
+                '(CASE WHEN ut.due_list_date_anc_1 BETWEEN %(start_date)s AND %(end_date)s THEN 1 ELSE 0 END) + '
+                '(CASE WHEN ut.due_list_date_anc_2 BETWEEN %(start_date)s AND %(end_date)s THEN 1 ELSE 0 END) + '
+                '(CASE WHEN ut.due_list_date_anc_3 BETWEEN %(start_date)s AND %(end_date)s THEN 1 ELSE 0 END) + '
+                '(CASE WHEN ut.due_list_date_anc_4 BETWEEN %(start_date)s AND %(end_date)s THEN 1 ELSE 0 END) '
+                ')'),
+            ('anc_1', 'ut.due_list_date_anc_1'),
+            ('anc_2', 'ut.due_list_date_anc_2'),
+            ('anc_3', 'ut.due_list_date_anc_3'),
+            ('anc_4', 'ut.due_list_date_anc_4'),
+            ('tt_1', 'ut.due_list_date_tt_1'),
+            ('tt_2', 'ut.due_list_date_tt_2'),
+            ('immediate_breastfeeding', 'agg_bp.immediate_breastfeeding'),
+            ('anemia', 'agg_bp.anemia'),
+            ('eating_extra', 'agg_bp.eating_extra'),
+            ('resting', 'agg_bp.resting'),
+            ('anc_weight', 'agg_bp.anc_weight'),
+            ('anc_blood_pressure', 'agg_bp.anc_blood_pressure'),
+            ('bp_sys', 'agg_bp.bp_sys'),
+            ('bp_dia', 'agg_bp.bp_dia'),
+            ('anc_hemoglobin', 'agg_bp.anc_hemoglobin'),
+            ('bleeding', 'agg_bp.bleeding'),
+            ('swelling', 'agg_bp.swelling'),
+            ('blurred_vision', 'agg_bp.blurred_vision'),
+            ('convulsions', 'agg_bp.convulsions'),
+            ('rupture', 'agg_bp.rupture'),
+            ('bp_date', 'agg_bp.latest_time_end_processed::DATE'),
+            ('is_ebf', 'agg_pnc.is_ebf'),
+            ('breastfed_at_birth', 'agg_delivery.breastfed_at_birth'),
+            ('person_name', 'case_list.person_name'),
+            ('edd', 'case_list.edd'),
+            ('delivery_nature', 'case_list.delivery_nature'),
+            ('mobile_number', 'case_list.mobile_number'),
+            ('preg_order', 'case_list.preg_order'),
+            ('num_pnc_visits', 'case_list.num_pnc_visits'),
+        )
+        return """
+        INSERT INTO "{tablename}" (
+            {columns}
+        ) (SELECT
+            {calculations}
+            FROM "{ucr_ccs_record_monthly_table}" ucr
+            LEFT OUTER JOIN "{agg_thr_table}" agg_thr ON ucr.doc_id = agg_thr.case_id AND ucr.month = agg_thr.month and ucr.valid_in_month = 1
```

Comment:
The `ucr.valid_in_month = 1` clause looks a bit strange to me. Does this function the same as a where clause? 

It's not a pattern I've been following so I'm not sure about the exact semantics. I've been defaulting to having those sorts of conditions in the calculations above, but this may make some of those cleaner